### DISCUSSION TITLE
能否多加一些和OS互动的功能

### DISCUSSION THREAD

**Discussion by cyhhao** [tw93/Pake#377] [Ideas]
# 功能请求
- 点击链接直接使用默认浏览器打开。现在是点了没反应。
- 下载文件自动下载到默认Download 文件夹。现在是下载到桌面的。
- 可设置关闭按钮的行为，退出 app 或是只关闭 window，不退出。

**Comment by Tlntin** [comment]
第一条：很早之前是这么干的，后面有需求就是部分链接内部打开不跳转到浏览器，结果改着改着就改成现在这样了。
第二条：很早之前也是下载到“下载”文件夹，后面不知道为啥改了很多次，结果改成桌面了。
第三条：这个关闭后还能再开，需要菜单栏托盘，实现起来问题不大，dev分支其实就实现了，不过要能设置行为，是关闭还是后台，相对就比较麻烦。

**Reply by tw93** [reply]
1. 其实也是跳出文件夹，这一块哈哈没有怎么改，是不是原本链接的跳转方式不是A标签的方式
2. 现在最新应该是 下载文件夹，我们cli发布一波应该就好啦

**Reply by cyhhao** [reply]
👍

第三点其实还有些更多的想法，哈哈，主要看这个项目的定位是想走极致简洁路线呢，还是配置化程度高的路线。

我感觉可以搞两个版本，一个Simple 一个 Advanced ，cli 也可以 `--advanced` 生成高级版本。
Simple 版本就维持现状
Advanced 版本多一个 Preferences 的面板，打开之后可以设置：
- 设置关闭按钮的行为
- macOS 点击托盘可以打开Pop-up 窗口，变成移动端的窗口快捷使用
- 设置快捷键快速唤出窗口
- 修改默认下载目录
- a标签打开链接方式，app内打开/chrome打开
- 自定义注入脚本

你们看 Advanced 版本的这个想法可以吗？可以的话我抽空试着搞一下

**Reply by Tlntin** [reply]
支持，欢迎pr .
